flink-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/KillerWatchDog.java

```diff
@@ -42,12 +42,12 @@
 	@SuppressWarnings("deprecation")
 	@Override
 	public void run() {
-		final long deadline = System.currentTimeMillis() + timeout;
+		final long deadline = System.nanoTime() + timeout * 1_000_000L;
 		long now;
 
-		while (toKill.isAlive() && (now = System.currentTimeMillis()) < deadline) {
+		while (toKill.isAlive() && (now = System.nanoTime()) < deadline) {
 			try {
-				toKill.join(deadline - now);
+				toKill.join( ( deadline - now ) / 1_000_000L );
```

Comment:
nit: I don't think we usually have spaces after braces like this here.
There isn't an agreed codestyle for this, but I think we can remove the spaces to be consistent with other parts of the code.

Will fix will merging.